### ISSUE TITLE
MD029/ol-prefix: Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]markdownlint(MD029)

### ISSUE THREAD

**Issue by xueqing** [DavidAnson/markdownlint#226]
I try to insert some code between ordered list items. The preview page is normal, but the raw doc reminds me to reset item prefix to 1. And my editor is VSCode 1.38.1.

 Example(test.md):

    # test order list

    1. step 1
    2. step 2

    ```sh
    echo "hello"
    ```

    1. step 3

# test order list

1. step 1
2. step 2

```sh
echo "hello"
```

3. step 3

![image](https://user-images.githubusercontent.com/6188285/66698098-a760af80-ed0d-11e9-8b46-782571a32461.png)

**Comment by nschonni** [comment]
The code fence needs to be indented under the second item, otherwise 2 lists are created and use a `start` value for the numbering on the second. IMO Markdownlint is correctly flagging this

**Comment by DavidAnson** [comment]
@nschonni is right. You can see the two OL groups for your top example here: https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%23%20test%20order%20list%5Cn%5Cn1.%20step%201%5Cn2.%20step%202%5Cn%5Cn%60%60%60sh%5Cnecho%20%5C%22hello%5C%22%5Cn%60%60%60%5Cn%5Cn1.%20step%203%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22src%22%7D%7D

**Comment by xueqing** [comment]
Thanks for your explanation! It works for me. And you may close the issue right now or should I close it?

**Comment by DavidAnson** [comment]
I will - thank you!

**Comment by harapeko** [comment]
https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%23%20test%20order%20list%5Cn%5Cn1.%20step%201%5Cn2.%20step%202%5Cn%5Cn%60%60%60sh%5Cnecho%20%5C%22hello%5C%22%5Cn%60%60%60%5Cn%5Cn1.%20step%203%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22src%22%7D%7D

![Image](https://github.com/user-attachments/assets/d214e547-7767-4a31-9a6b-f0da99765021)
